src/spaces.hpp

```diff
@@ -248,6 +248,14 @@ struct vector : entity<vector<T, N>> {
 template <class... Ts>
 vector(Ts...) -> vector<std::common_type_t<Ts...>, sizeof...(Ts)>;
 
+template <class, std::size_t>
+struct extend_to {};
+
+template <Arithmetic T, std::size_t N, std::size_t NewN>
+struct extend_to<vector<T, N>, NewN> {
+    using type = vector<T, NewN>;
+};
```

Comment:
Can you add a convenience type alias so it's easier to read when used?
```
template  <...>
using extend_to_t = typename extend_to<...>::type;
```

Reply:
Done